avocado/utils/distro.py

```diff
@@ -331,6 +331,40 @@ class FedoraProbe(RedHatProbe):
     CHECK_VERSION_REGEX = re.compile(r'Fedora release (\d{1,2}).*')
 
 
+class AmazonLinuxProbe(Probe):
+
+    """
+    Probe for Amazon Linux systems
+    """
+
+    CHECK_FILE = '/etc/os-release'
+    CHECK_FILE_CONTAINS = 'Amazon Linux'
+    CHECK_FILE_DISTRO_NAME = 'Amazon Linux'
```

Comment:
Not that there's a rule here, but other distros probes use a lower case, single word for the distro name (rhel, fedora, centos).  I'm not sure if it's applicable to Amazon Linux, though.  I see:

```
ID="amzn"
ID_LIKE="rhel fedora"
```

On `/etc/os-release`, which makes me think that maybe this is the distro short name... but I'm only guessing.

Reply:
Sure, but I'm not the biggest fan of 'amzn', I must say. If that's the expected unwritten rule, I can change the code.

Reply:
cat /etc/os-release
    NAME="Amazon Linux"
    VERSION="2 (2017.12)"
    ID="amzn"
    ID_LIKE="centos rhel fedora"
    VERSION_ID="2"
    PRETTY_NAME="Amazon Linux 2 (2017.12) LTS Release Candidate"
    ANSI_COLOR="0;33"
    CPE_NAME="cpe:2.3:o:amazon:amazon_linux:2"
    HOME_URL="https://amazonlinux.com/"

Reply:
cat /etc/os-release 
    NAME="Amazon Linux AMI"
    VERSION="2018.03"
    ID="amzn"
    ID_LIKE="rhel fedora"
    VERSION_ID="2018.03"
    PRETTY_NAME="Amazon Linux AMI 2018.03"
    ANSI_COLOR="0;33"
    CPE_NAME="cpe:/o:amazon:linux:2018.03:ga"
    HOME_URL="http://aws.amazon.com/amazon-linux-ami/"

Reply:
I'm open to suggestions. We can make it 'amazon_linux', 'AmazonLinux', etc. I just don't think 'amzn' is too descriptive, although sure, you can argue that this soup of letters can be ultimately associated with the company.

Reply:
Lowercase `al` seems vague. I don't know.

Reply:
Yeah, I don't have a better suggestion, and naming things is *hard*.  In the end it's your call.  Let me know if we should go ahead with the originally proposed name.